### PR TITLE
Add Github Enterprise support

### PR DESCRIPTION
**Pull request by lex-ibm**
Adds support for using this repo in a private Github Enterprise instance.

### CODE REVIEW

**Comment by lex-ibm** [comment]
I did not include the `SHOULD_BUILD_TAR="${SHOULD_BUILD_TAR:-no}"` changes. I don't think those affect the current build process but given the script is not meant to be used like that I don't see the case.

File: check_tags.sh [hunk 1]
```diff
@@ -3,17 +3,20 @@
 
 set -e
 
-if [[ -z "${GITHUB_TOKEN}" ]]; then
+if [[ -z "${GH_TOKEN}" ]] && [[ -z "${GITHUB_TOKEN}" ]] && [[ -z "${GH_ENTERPRISE_TOKEN}" ]] && [[ -z "${GITHUB_ENTERPRISE_TOKEN}" ]]; then
```
Comment (daiyam) [inline check_tags.sh:6]: Is it missing the `else` statement like in `update_version.sh`?

**Comment by daiyam** [comment]
> I did not include the `SHOULD_BUILD_TAR="${SHOULD_BUILD_TAR:-no}"` changes. I don't think those affect the current build process but given the script is not meant to be used like that I don't see the case.

Yep, it doesn't affect the VSCodium's build process. So it's up to you.

File: check_tags.sh (hunk 1 above)
Comment (lex-ibm) [reply check_tags.sh:6]: It was. Changed it to match.

**Comment by lex-ibm** [comment]
I'll change my pipeline instead. I don't want to change how `check_tags.sh` works other than adding GitHub Enterprise support.

**Comment by daiyam** [comment]
LGTM

**Comment by daiyam** [comment]
OH! A last thought, wouldn't `GITHUB_HOST` be better than `GH_HOST`?

**Comment by daiyam** [comment]
> OH! A last thought, wouldn't `GITHUB_HOST` be better than `GH_HOST`?

Can you make the change?

**Comment by lex-ibm** [comment]
> > OH! A last thought, wouldn't `GITHUB_HOST` be better than `GH_HOST`?
> 
> Can you make the change?

`GH_HOST` comes from the [GitHub CLI documentation](https://cli.github.com/manual/gh_help_environment).

**Comment by lex-ibm** [comment]
Changing `GH_HOST` to `GITHUB_HOST` would mean changing the calls to `gh` adding the `--hostname` flag.

**Comment by daiyam** [comment]
@lex-ibm Thanks

**Comment by lex-ibm** [comment]
@daiyam thank you! I'll work on getting those electron binaries out.

**Pull request merged**